### PR TITLE
SCF-Session-8-Assignment Commit for submission

### PR DESCRIPTION
**Pull request by meta-jugal-kukreja**

### CODE REVIEW

File: SCF-Session-8-Assignment/src/question/Animal.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+	 public double animalWeight;
+	 public int animalAge;
+	 public String animalType;
+	 
+	 public abstract String getSound();
+	 
+	 /**
+	  * The constructor only increments the id whenever its object is created
+	  */
+	 Animal()
```
Comment (meta-deepak-goyal) [inline SCF-Session-8-Assignment/src/question/Animal.java]: Initialize all the property of Animal class in the parameterized constructor.

File: SCF-Session-8-Assignment/src/question/Bird.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+ * This class is for category Birds
+ * @author Jugal
+ * Dated 07/23/2019
+ */
+public abstract class Bird extends Animal{
+
+	/**
+	 * Constructor to assign specialty and category
+	 */
+	public Bird()
```
Comment (meta-deepak-goyal) [inline SCF-Session-8-Assignment/src/question/Bird.java]: Declare some different property of the Bird class, and initialize all the property of Bird and Animal class using parameterized constructor and super().

File: SCF-Session-8-Assignment/src/question/Mammal.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
+package question;
+
+/**
+ * This class is for category Mammal
+ * @author Jugal
+ * Dated 07/23/2019
+ */
+public abstract class Mammal extends Animal{
+
```
Comment (meta-deepak-goyal) [inline SCF-Session-8-Assignment/src/question/Mammal.java]: Declare some different properties of Mammal class and initialize them in Mammal class constructor and initialize Animal class property using super().

File: SCF-Session-8-Assignment/src/question/Lion.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+public class Lion extends Mammal {
+
+	/**
+	 * Constructor to initialize the object
+	 * 
+	 * @param name is the name of the animal
+	 * @param type is the type of the animal
+	 * @param category is the category of the animal
+	 */
+	public Lion(String name, String type, String category) {
```
Comment (meta-deepak-goyal) [inline SCF-Session-8-Assignment/src/question/Lion.java]: Declare some different properties of Lion class and initialize them in Lion class constructor and initialize Mammal and  Animal class property using super().

File: SCF-Session-8-Assignment/src/question/Peacock.java [hunk 5]
```diff
@@ -0,0 +1,28 @@
+package question;
+
+/**
+ * This class is for Peacock type animals
+ * @author Jugal
+ * Dated 07/24/2019
+ */
+public class Peacock extends Bird {
+
+	/**
```
Comment (meta-deepak-goyal) [inline SCF-Session-8-Assignment/src/question/Peacock.java]: Declare some different properties of Peacock class and initialize them in Peacock class constructor and initialize Bird and Animal class property using super().

File: SCF-Session-8-Assignment/src/question/Reptile.java [hunk 6]
```diff
@@ -0,0 +1,18 @@
+package question;
+
+/**
+ * This class is for category Reptile
+ * @author Jugal
+ * Dated 07/23/2019
+ */
+public abstract class Reptile extends Animal {
+
```
Comment (meta-deepak-goyal) [inline SCF-Session-8-Assignment/src/question/Reptile.java]: Declare some different properties of Reptile class and initialize them in Reptile class constructor and initialize Animal class property using super().

File: SCF-Session-8-Assignment/src/question/Snake.java [hunk 7]
```diff
@@ -0,0 +1,28 @@
+package question;
+
+/**
+ * This class is for Snake type animals
+ * @author Jugal
+ * Dated 07/24/2019
+ */
+public class Snake extends Reptile {
+
```
Comment (meta-deepak-goyal) [inline SCF-Session-8-Assignment/src/question/Snake.java]: Declare some different properties of Snake class and initialize them in Snake class constructor and initialize Reptile and Animal class property using super().

File: SCF-Session-8-Assignment/src/question/Cage.java [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+import java.util.List;
+
+/**
+ * This class is for the cages in the zones
+ * @author Jugal
+ * Dated 07/24/2019
+ */
+public class Cage {
+	
+	 int cageCapacity;
```
Comment (meta-deepak-goyal) [inline SCF-Session-8-Assignment/src/question/Cage.java]: Declare the variables using access modifiers.

File: SCF-Session-8-Assignment/src/question/Zone.java [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+import java.util.List;
+
+/**
+ * This Class is for the zones in the ZOO. The zones contains cages as a ArrayList in the ZOO.
+ * @author Jugal
+ * Dated 07/24/2019
+ */
+public class Zone {
+	
+	static int zoneId = 0;
```
Comment (meta-deepak-goyal) [inline SCF-Session-8-Assignment/src/question/Zone.java]: Declare the variables using access modifiers.

File: SCF-Session-8-Assignment/src/question/Zoo.java [hunk 10]
```diff
@@ -0,0 +1,195 @@
...
+	 * @param animalType is the type of that animal name
+	 * @return {Animal} animal object
+	 */
+	public Animal buildAnimalObject(String animalName, String animalType)
+	{
+		Animal animalObj = null;
+		switch (animalType.toLowerCase())
+		{
+		case "lion":
+			animalObj = new Lion(animalName, animalType, Lion.class.getSuperclass().getSimpleName());
```
Comment (meta-deepak-goyal) [inline SCF-Session-8-Assignment/src/question/Zoo.java]: get the category using getCatgory() method you declared.

File: SCF-Session-8-Assignment/src/question/Zoo.java [hunk 11]
```diff
@@ -0,0 +1,195 @@
...
+			break;
+		
+		case "peacock":
+			animalObj = new Peacock(animalName, animalType, Peacock.class.getSuperclass().getSimpleName());
```
Comment (meta-deepak-goyal) [inline SCF-Session-8-Assignment/src/question/Zoo.java]: get the category using getCatgory() method you declared.

File: SCF-Session-8-Assignment/src/question/Zoo.java [hunk 12]
```diff
@@ -0,0 +1,195 @@
...
+			break;
+			
+		case "snake":
+			animalObj = new Snake(animalName, animalType, Snake.class.getSuperclass().getSimpleName());
```
Comment (meta-deepak-goyal) [inline SCF-Session-8-Assignment/src/question/Zoo.java]: get the category using getCatgory() method you declared.

File: SCF-Session-8-Assignment/src/question/Animal.java (hunk 1 above)
Comment (meta-jugal-kukreja) [reply SCF-Session-8-Assignment/src/question/Animal.java]: Done

File: SCF-Session-8-Assignment/src/question/Bird.java (hunk 2 above)
Comment (meta-jugal-kukreja) [reply SCF-Session-8-Assignment/src/question/Bird.java]: Done

File: SCF-Session-8-Assignment/src/question/Mammal.java (hunk 3 above)
Comment (meta-jugal-kukreja) [reply SCF-Session-8-Assignment/src/question/Mammal.java]: Done

File: SCF-Session-8-Assignment/src/question/Lion.java (hunk 4 above)
Comment (meta-jugal-kukreja) [reply SCF-Session-8-Assignment/src/question/Lion.java]: Done

File: SCF-Session-8-Assignment/src/question/Peacock.java (hunk 5 above)
Comment (meta-jugal-kukreja) [reply SCF-Session-8-Assignment/src/question/Peacock.java]: Done

File: SCF-Session-8-Assignment/src/question/Reptile.java (hunk 6 above)
Comment (meta-jugal-kukreja) [reply SCF-Session-8-Assignment/src/question/Reptile.java]: Done

File: SCF-Session-8-Assignment/src/question/Snake.java (hunk 7 above)
Comment (meta-jugal-kukreja) [reply SCF-Session-8-Assignment/src/question/Snake.java]: Done

File: SCF-Session-8-Assignment/src/question/Cage.java (hunk 8 above)
Comment (meta-jugal-kukreja) [reply SCF-Session-8-Assignment/src/question/Cage.java]: Given the package access for the variable.

File: SCF-Session-8-Assignment/src/question/Zone.java (hunk 9 above)
Comment (meta-jugal-kukreja) [reply SCF-Session-8-Assignment/src/question/Zone.java]: Given the package access for the variable.

File: SCF-Session-8-Assignment/src/question/Zoo.java (hunk 10 above)
Comment (meta-jugal-kukreja) [reply SCF-Session-8-Assignment/src/question/Zoo.java]: Done.
Changed the logic.

File: SCF-Session-8-Assignment/src/question/Zoo.java (hunk 11 above)
Comment (meta-jugal-kukreja) [reply SCF-Session-8-Assignment/src/question/Zoo.java]: Done.
Changed the logic.

File: SCF-Session-8-Assignment/src/question/Zoo.java (hunk 12 above)
Comment (meta-jugal-kukreja) [reply SCF-Session-8-Assignment/src/question/Zoo.java]: Done.
Changed the logic.

**Pull request merged**